src/quota_aggregator_impl.cc

```diff
@@ -151,36 +149,45 @@ ::google::protobuf::util::Status QuotaAggregatorImpl::CacheResponse(
   AllocateQuotaCacheRemovedItemsHandler::StackBuffer::Swapper swapper(
       this, &stack_buffer);
 
-  if (cache_) {
-    string request_signature = GenerateAllocateQuotaRequestSignature(request);
-    QuotaCache::ScopedLookup lookup(cache_.get(), request_signature);
+  InternalCacheResponse(request, response, false);
+
+  return ::google::protobuf::util::Status::OK;
+}
+
+// This method must be called behind the cache_mutex_ lock
+void QuotaAggregatorImpl::InternalCacheResponse(
+    const ::google::api::servicecontrol::v1::AllocateQuotaRequest& request,
+    const ::google::api::servicecontrol::v1::AllocateQuotaResponse& response,
+    bool is_refreshing) {
+  if (!cache_) {
+    return;
+  }
 
-    CacheElem* cache_elem = new CacheElem(response, SimpleCycleTimer::Now());
-    cache_elem->set_signature(request_signature);
+  string request_signature = GenerateAllocateQuotaRequestSignature(request);
```

Comment:
I have removed the recalculation and unnecessary unnecessary request argument afected by this.
Changed the prototype
  void CacheResponseWithLock(
      const std::string& request_signature,
      const ::google::api::servicecontrol::v1::AllocateQuotaResponse& response);